extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java

```diff
@@ -126,51 +123,11 @@ public IntBuffer readKeyValues(IntBuffer keyDest, int nullPlaceholder,
         try (
                 final ContextHolder holder = SeekableChannelContext.ensureContext(channelsProvider, channelContext);
                 final SeekableByteChannel ch = channelsProvider.getReadChannel(holder.get(), uri)) {
-            ensurePageHeader(channelsProvider, ch);
+            ch.position(dataOffset);
             return readKeysFromDataPage(keyDest, nullPlaceholder, ch, holder.get());
         }
     }
 
-    /**
-     * If {@link #pageHeader} is {@code null}, read it from the channel, and increment the {@link #offset} by the length
-     * of page header. Channel position would be set to the end of page header or beginning of data before returning.
-     */
-    private void ensurePageHeader(SeekableChannelsProvider provider, SeekableByteChannel ch) throws IOException {
-        // Set this channel's position to appropriate offset for reading. If pageHeader is null, this offset would be
-        // the offset of page header, else it would be the offset of data.
-        ch.position(offset);
-        synchronized (this) {
-            if (pageHeader == null) {
-                try (final InputStream in = SeekableChannelsProvider.channelPositionInputStream(provider, ch)) {
-                    pageHeader = Util.readPageHeader(in);
-                }
-                offset = ch.position();
-                if (numValues >= 0) {
-                    final int numValuesFromHeader = readNumValuesFromPageHeader(pageHeader);
-                    if (numValues != numValuesFromHeader) {
-                        throw new IllegalStateException(
-                                "numValues = " + numValues + " different from number of values " +
-                                        "read from the page header = " + numValuesFromHeader + " for column " + path);
-                    }
-                }
-            }
-            if (numValues == NULL_NUM_VALUES) {
-                numValues = readNumValuesFromPageHeader(pageHeader);
-            }
-        }
-    }
-
-    private static int readNumValuesFromPageHeader(@NotNull final PageHeader header) throws IOException {
-        switch (header.type) {
-            case DATA_PAGE:
-                return header.getData_page_header().getNum_values();
-            case DATA_PAGE_V2:
-                return header.getData_page_header_v2().getNum_values();
-            default:
-                throw new IOException(String.format("Unexpected page of type {%s}", header.getType()));
-        }
-    }
```

Comment:
No need for all this because this is already done at the time of constructing this reader object.